python/pyspark/sql/connect/client.py

```diff
@@ -626,6 +627,10 @@ def _handle_error(self, rpc_error: grpc.RpcError) -> NoReturn:
                         raise SparkConnectTempTableAlreadyExistsException(
                             info.metadata["message"], plan=info.metadata["plan"]
                         ) from None
+                    elif reason == "java.lang.IllegalArgumentException":
```

Comment:
Can we deduplicate this w/ `convert_exception` at `pyspark.errors.exceptions`? You can do it in a followup.